apm-agent-core/src/main/java/co/elastic/apm/agent/impl/ElasticApmTracerBuilder.java

```diff
@@ -100,8 +105,13 @@ public ElasticApmTracer build() {
             final List<ConfigurationSource> configSources = getConfigSources(agentArguments);
             configurationRegistry = getDefaultConfigurationRegistry(configSources);
         }
+        final ApmServerClient apmServerClient = new ApmServerClient(configurationRegistry.getConfig(ReporterConfiguration.class));
+        final DslJsonSerializer payloadSerializer = new DslJsonSerializer(configurationRegistry.getConfig(StacktraceConfiguration.class));
+        final MetaData metaData = MetaData.create(configurationRegistry, null, null);
+        configurationRegistry.addConfigurationSource(new ApmServerConfigurationSource(payloadSerializer, metaData, apmServerClient));
+        configurationRegistry.reloadDynamicConfigurationOptions();
```

Comment:
This block the startup until there's a response from the APM Server.

Pros:
- Does not start the agent without applying the remote configuration
  Avoids that the agent starts with 100% sampling and only bring it down to the remotely configured level after 30s
- We could also load non-dynamic/reloadable options remotely

Reply:
That should be aligned with the other agents. I've added a comment, listing the options here: https://github.com/elastic/apm/issues/76#issuecomment-509709435

Reply:
Added a comment too, tried to push this to a conclusion.